src/csharp/Microsoft.Spark.E2ETest/UdfTests/UdfComplexTypesTests.cs

```diff
@@ -197,15 +197,15 @@ public void TestUdfWithRowType()
         [Fact]
         public void TestUdfWithReturnAsRowType()
         {
-            // Single GenericRow
+            // Single Row.
             {
                 var schema = new StructType(new[]
                 {
                     new StructField("col1", new IntegerType()),
                     new StructField("col2", new StringType())
                 });
                 Func<Column, Column> udf = Udf<string>(
-                    str => new GenericRow(new object[] { 1, "abc" }), schema);
+                    str => new Row(new object[] { 1, "abc" }), schema);

```

Comment:
Can we allow GenericRow to be returned as well?

So I am thinking the user can either do
```C#
 Func<Column, Column> udf = Udf<string>(
    str => new GenericRow(new object[] { 1, "abc" }), schema);
```
or
```C#
 Func<Column, Column> udf = Udf<string>(
    str => new Row(new object[] { 1, "abc" }, schema), schema);
```
Can you check if the latter example works? (to see if schema is serialized correctly)

Reply:
You probably need this: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/user-defined-conversion-operators

Reply:
The `Row` constructor for the following example will also be internal, right? Same point as your below comment.
```csharp
 Func<Column, Column> udf = Udf<string>(
    str => new Row(new object[] { 1, "abc" }, schema), schema);
```

Reply:
Actually, let's be conservative and just allow only the following:
```C#
 Func<Column, Column> udf = Udf<string>(
    str => new GenericRow(new object[] { 1, "abc" }), schema);
```
, and make the user defined conversion operator as internal as well.

The argument is that we don't want to allow users to create a schemaless `Row`. (We should also comment that schemaless Rows can happen within chained UDFs (same behavior as PySpark).)

Reply:
Looks like I cannot make user-defined operator as internal.

After adding the user-defined conversion operator, the following works:
```csharp
GenericRow genericRow = new GenericRow(new object[] { 111 });
Row row = (Row)genericRow;
```